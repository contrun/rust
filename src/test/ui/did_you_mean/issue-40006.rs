impl dyn A {
    Y //~ ERROR non-item in item list
}

struct S;

trait X {
    X() {} //~ ERROR non-item in item list
    fn xxx() { ### }
    L = M;
    Z = { 2 + 3 };
    ::Y ();
}

trait A {
    X() {} //~ ERROR non-item in item list
}
trait B {
    fn xxx() { ### } //~ ERROR expected
}
trait C {
    L = M; //~ ERROR non-item in item list
}
trait D {
    Z = { 2 + 3 }; //~ ERROR non-item in item list
}
trait E {
    ::Y (); //~ ERROR non-item in item list
}

impl S {
    pub hello_method(&self) { //~ ERROR missing
        println!("Hello");
    }
}

fn main() {
    S.hello_method(); //~ no method named `hello_method` found
}
