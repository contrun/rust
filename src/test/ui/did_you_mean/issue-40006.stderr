error: non-item in item list
  --> $DIR/issue-40006.rs:2:5
   |
LL | impl dyn A {
   |            - item list starts here
LL |     Y
   |     ^ non-item starts here
LL | }
   | - item list ends here

error: non-item in item list
  --> $DIR/issue-40006.rs:8:5
   |
LL | trait X {
   |         - item list starts here
LL |     X() {}
   |     ^ non-item starts here
...
LL | }
   | - item list ends here

error: non-item in item list
  --> $DIR/issue-40006.rs:16:5
   |
LL | trait A {
   |         - item list starts here
LL |     X() {}
   |     ^ non-item starts here
LL | }
   | - item list ends here

error: expected `[`, found `#`
  --> $DIR/issue-40006.rs:19:17
   |
LL |     fn xxx() { ### }
   |                 ^ expected `[`

error: non-item in item list
  --> $DIR/issue-40006.rs:22:5
   |
LL | trait C {
   |         - item list starts here
LL |     L = M;
   |     ^ non-item starts here
LL | }
   | - item list ends here

error: non-item in item list
  --> $DIR/issue-40006.rs:25:5
   |
LL | trait D {
   |         - item list starts here
LL |     Z = { 2 + 3 };
   |     ^ non-item starts here
LL | }
   | - item list ends here

error: non-item in item list
  --> $DIR/issue-40006.rs:28:5
   |
LL | trait E {
   |         - item list starts here
LL |     ::Y ();
   |     ^^ non-item starts here
LL | }
   | - item list ends here

error: missing `fn` for method definition
  --> $DIR/issue-40006.rs:32:8
   |
LL | impl S {
   |        - while parsing this item list starting here
LL |     pub hello_method(&self) {
   |        ^
...
LL | }
   | - the item list ends here
   |
help: add `fn` here to parse `hello_method` as a public method
   |
LL |     pub fn hello_method(&self) {
   |         ^^

error[E0599]: no method named `hello_method` found for struct `S` in the current scope
  --> $DIR/issue-40006.rs:38:7
   |
LL | struct S;
   | --------- method `hello_method` not found for this
...
LL |     S.hello_method();
   |       ^^^^^^^^^^^^ method not found in `S`

error: aborting due to 9 previous errors

For more information about this error, try `rustc --explain E0599`.
